sale_stock_cancel_restriction/tests/test_sale_stock_cancel_restriction.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2021 Tecnativa - Ernesto Tejeda
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo.exceptions import UserError
+from odoo.tests import Form, SavepointCase
+
+
+class TestSaleStockCancelRestriction(SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.product = cls.env["product.product"].create(
+            {"name": "Product test", "type": "product"}
+        )
+        cls.partner = cls.env["res.partner"].create({"name": "Partner test"})
+        so_form = Form(cls.env["sale.order"])
+        so_form.partner_id = cls.partner
+        with so_form.order_line.new() as soline_form:
+            soline_form.product_id = cls.product
+            soline_form.product_uom_qty = 2
+        cls.sale_order = so_form.save()
+        cls.sale_order.action_confirm()
+        cls.picking = cls.sale_order.picking_ids
+        cls.picking.move_lines.quantity_done = 2
+
+    def test_cancel_sale_order_restrict(self):
+        """
+        Validates the picking and do the assertRaises cancelling the
+        order for checking that it's forbidden
+        :return:
+        """
+        self.picking.button_validate()
+        with self.assertRaises(UserError):
+            self.sale_order.action_cancel()
+
+    def test_cancel_sale_order_ok(self):
+        """
+        Don't validate the picking and cancel the order, being completed.
+        """
```

Comment:
If it fits in one line, the docstring is:

```suggestion
        """Don't validate the picking and cancel the order, being completed."""
```

but if not, the first line goes with the opening quotes.